building/configuring_git_workflow.adoc

```diff
@@ -0,0 +1,426 @@
+
+image::../images/import-maven-project-5-tycho-pde.png[]
+
+=== How to add forks to your local clone
+
+If you use https://hub.github.com/[hub], you can simply add a fork like this:
+
+[source,bash]
+----
+cd ~/my-github-stuff/jbosstools-base
+git remote add -p ${gituser}
+----
+
+Or, without https://hub.github.com/[hub], do this:
+
+[source,bash]
+----
+cd ~/my-github-stuff/jbosstools-base
+git remote add ${gituser} git@github.com:${gituser}/jbosstools-base.git
+----
+
+Then to check which forks you have attached:
+
+[source,bash]
+----
+git remove -v
+
+nickboldt   git@github.com:nickboldt/jbosstools-devdoc.git (fetch)
+nickboldt   git@github.com:nickboldt/jbosstools-devdoc.git (push)
+origin  git@github.com:jbosstools/jbosstools-devdoc.git (fetch)
+origin  git@github.com:jbosstools/jbosstools-devdoc.git (push)
+----
+
+
+== Submitting Pull Requests as a Contributor
+
+It is recommended that you use https://hub.github.com/[hub] to simplify your commandline git experience.
+
+To install hub, go here: https://hub.github.com/
+
+=== Pull Request Guidelines
+
+The rules are simple:
+
+. Ensure you've pulled the latest from the upstream origin's fork and topic branch
+. Rebase/merge as needed
+. Topic branch should contain JIRA ID
+. Commit comment(s) should also contain JIRA ID
+
+[source,bash]
+----
+# checkout the topic branch and origin branch, and link them
+# ${topic} = topic branch in your fork
+# ${branch} = branch in the origin
+git checkout origin/${branch} -b ${topic}
+git checkout ${topic}
+
+# commit changes
+git commit -m "JBIDE-12345 fix that thing that was broken" somefile
+git commit -m "JBIDE-12345 additional fix" somefile
+git commit -m "typo" somefile
+
+# squash commits?
+git rebase -i HEAD~3 # eg., if you want to squash 3 commits into a single one
+
+# rebase your changes against the origin
+git pull --rebase origin ${branch}
+
+# push to your fork
+# ${gituser} is your username at github
+git push ${gituser} ${topic}
+
+# create pull request
+# if you use hub (see `Submitting Patches` above), you can create a pull request via commandline:
+parentProject=`git remote -v | grep origin | grep push | sed "s/.\+github.com\(:\|\/\)\(.\+\)\/.\+/\2/"`
+hub pull-request -o -f -m "`git log -1 --pretty=%B` >> ${branch}" -b ${parentProject}:${branch} -h ${gituser}:${topic}
+----
+
+== Accepting Patches (Maintainers)
+
+Maintainers and committers are required to do due dilligence on any patches that come their way. 
+This includes an initial *code review* to catch any obvious problems, as well as *smoke tests*.
+Maintainers may require patches come with *unit tests* as well, so they can easily verify that the
+given patch does in fact fix the given issue. 
+
+To check out a given contribution and run the tests or manually smoke-test the contribution, 
+maintainers should look at the pull request to determine 
+the contributor's GitHub username, as well as the topic branch. 
+
+```bash
+cd jbosstools-server
```

Comment:
I think you mean...

```
Or, when using hub:

```bash
cd jbosstools-server
git co https://github.com/jbosstools/jbosstools-server/pull/999
```
```